node/src/components/linear_chain_sync/peers.rs

```diff
@@ -26,7 +36,12 @@ impl<I: Clone + PartialEq + 'static> PeersState<I> {
 
     /// Returns a random peer.
     pub(crate) fn random(&mut self) -> Option<I> {
-        self.peers_to_try.pop()
+        if self.succ_attempts < self.succ_attempts_max {
+            self.next_succ().or_else(|| self.peers_to_try.pop())
+        } else {
+            self.succ_attempts = 0;
+            self.peers_to_try.pop()
```

Comment:
If there are no more `peers_to_try`, shouldn't we use one from `succ_peers` anyway?

Reply:
Oh, right. Good catch.

Reply:
Done in [`8b25d5d` (#784)](https://github.com/CasperLabs/casper-node/pull/784/commits/8b25d5d0e84fc319c6f387104c1885d09e610158)